task-tracker-console-application/src/com/isa/user/Login.java

```diff
@@ -0,0 +1,134 @@
+package com.isa.user;
+
+import com.isa.MainMenu.MainMenu;
+
+import java.util.List;
+import java.util.Scanner;
+
+public class Login {
+    private static User loggedInUser;
+    private static final List<User> users = User.getTestUsers();
+    private static String inputLogin;
+
+    public static User getLoggedInUser() {
+        return loggedInUser;
+    }
+
+    public static void logOutUser() {
+        loggedInUser = null;
+    }
+
+    private static void displayLoginMessage() {
+        System.out.println("=====================================");
+        System.out.println("             ZALOGUJ SIĘ");
+        System.out.println("=====================================");
+        System.out.println("Podaj dane logowania lub wybierz \"0\", żeby powrócić do menu głównego.");
+    }
+
+    private static void logInUser(Scanner scanner) {
+        System.out.print("Podaj login: ");
+        validateLogin(scanner, 0);
+    }
+
+    private static void validateLogin(Scanner scanner, int incorrectLoginInputCount) {
+        inputLogin = scanner.nextLine();
+        if (inputLogin.equals("0")) {
+            goToMainMenu();
+        } else if (chosenUser() == null) {
+            incorrectLoginInputCount++;
+            System.out.println("Podany login nie istnieje.");
+            if (incorrectLoginInputCount > 1) {
+                System.out.println("Pamiętaj, że w każdej chwili możesz wrócić do menu głównego poprzez wybór \"0\".");
+            }
+
+            System.out.print("Podaj login ponownie: ");
+            validateLogin(scanner, incorrectLoginInputCount);
+        } else {
+            System.out.print("Podaj hasło: ");
+            validatePassword(scanner, 0);
+        }
+    }
+
+    private static User chosenUser() {
+        for (User u : users) {
+            if (u.getLogin().equals(inputLogin)) {
+                return u;
+            }
+        }
+        return null;
+    }
+
+    private static void validatePassword(Scanner scanner, int incorrectPasswordInputCount) {
+        String inputPassword = scanner.nextLine();
+        if (inputPassword.equals("0")) {
+            goToMainMenu();
+        } else if (!isCorrectPassword(inputPassword)) {
+            incorrectPasswordInputCount++;
+            System.out.println("Hasło niepoprawne.");
+            if (incorrectPasswordInputCount > 1 && incorrectPasswordInputCount < 5) {
+                System.out.println("Pamiętaj, że w każdej chwili możesz wrócić do menu głównego poprzez wybór \"0\".");
+                if (incorrectPasswordInputCount == 4) {
+                    System.out.println("UWAGA: Przy kolejnej nieprawidłowej próbie zalogowania się twoje konto zostanie zablokowane.");
+                }
```

Comment:
trochę dużo poziomów tych zagnieżdżeń, można to wydzielić do osobnych funkcji

Reply:
Utworzyłam osobną metodę na sprawdzanie, ile razy podano niepoprawne hasło (checkForIncorrectPassword)